examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+
+public static void demoConnection(CastingPlayer castingPlayer) {
+    // Connect to the castingPlayer
+    CompletableFuture<Void> connectionFuture = castingPlayer.connect(CONNECTION_TIMEOUT_SECS);
+
+    // handle exception, if any
+    connectionFuture.exceptionally(exc -> {
+        Log.e(TAG, "Exception in connecting to castingPlayer" + exc.getMessage());
+        return null;
+    });
+}
+```
+
+On iOS, the Casting Client may call `connect` on the `MTRCastingPlayer` object
+it wants to connect to and handle any `NSErrors` that may happen in the process.
+
+```objectivec
+@Published var connectionSuccess: Bool?;
+
+@Published var connectionStatus: String?;
+
+func connect(selectedCastingPlayer: MTRCastingPlayer?) {
+    selectedCastingPlayer?.connect(withTimeout: 60, withCompletionBlock: { err in
+        self.Log.error("MTRConnectionExampleViewModel connect() completed with \(err)")
+        if(err == nil)
+        {
+            self.connectionSuccess = true
+            self.connectionStatus = "Connected!"
+        }
+        else
+        {
+            self.connectionSuccess = false
+            self.connectionStatus = "Connection failed with \(String(describing: err))"
+        }
+    })
+}
+```
+
+### Select an Endpoint on the Casting Player
+
+_{Endpoing selections is demonstrated in the Command invocation examples:
+[Android](android/App/app/src/main/java/com/matter/casting/CommandInvocationExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRCommandInvocationExampleViewModel.swift)}_
+
+On a successful connection with a `CastingPlayer`, a Casting Client may select
+one of the `Endpoints` to interact with based on its attributes (e.g. Vendor ID,
+Product ID, list of supported Clusters, etc).
+
+On Android, for example, it may select an Endpoint with a particular vendorID.
+
+```java
+...
+Optional<Endpoint> contentAppEndpoint =
+    castingPlayer
+        .getEndpoints()
+        .stream()
+        .filter(endpoint -> SAMPLE_CONTENT_APP_VID.equals(endpoint.getVendorId()))
+        .findAny();
+...
+```
+
+On iOS, it can do the above Endpoint selection like the following.
+
+```objectivec
+...
+if let endpoint: MTREndpoint = castingPlayer.endpoints.filter({ $0.identifier == sampleContentAppVid }).first
+...
+```
+
+## Interacting with a Casting Endpoint
+
+Once the Casting Client has selected an `Endpoint`, it is ready to
+[issue commands](#issuing-commands) to it, [read](#read-operations) current
+playback state, and [subscribe](#subscriptions) to playback events.
+
+### Issuing Commands
+
+_{Complete Command invocation examples:
+[Android](android/App/app/src/main/java/com/matter/casting/CommandInvocationExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRCommandInvocationExampleViewModel.swift)}_
+
+To issue a command, the Casting Client can `contentAppEndpoint` on the
+CastingPlayer it connected to and handle the command response. It would first
+ensure the endpoint supports the required cluster/command.
```

Comment:
This paragraph is not clear. Are you meaning to say the client can get a reference to the `contentAppEndpoint` and then send commands to it?

Reply:
Yes, thats what I meant. I have tried rephrasing this in the latest version of the diff.